src/pages/RoadObstructions/components/FiltersForm/Form.test.tsx

```diff
@@ -0,0 +1,195 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { format } from 'date-fns'
+
+import { withAppContext } from '../../../../../test/utils/withAppContext'
+
+import {
+  RoadObstructionsFiltersForm,
+  RoadObstructionsFiltersFormProps,
+} from './Form'
+
+const defaultFormValues = {
+  date: format(new Date(), 'yyyy-MM-dd'),
+  timeFrom: '00:00',
+  timeTo: '23:59',
+}
+
+describe('RoadObstructionsFiltersForm', () => {
+  const props: RoadObstructionsFiltersFormProps = {
+    mapFilters: {
+      date: '',
+      timeFrom: '',
+      timeTo: '',
+    },
+    setMapFilters: jest.fn(),
+    setShowMapFiltersForm: jest.fn(),
+  }
+
+  it('displays error messages when no input is provided', async () => {
+    render(withAppContext(<RoadObstructionsFiltersForm {...props} />))
+
+    fireEvent.submit(screen.getByRole('button'))
```

Comment:
Is er een reden om hier `fireEvent` ipv `userEvent` te gebruiken?

Reply:
Dank voor de review @janthijs :+1:  

Goede vraag. In eerste instantie heb ik geprobeerd dit form te testen mbv `userEvent`. De date input en submit button werkt zonder problemen, maar de time input is tot op heden een onopgeloste puzzel. Ik heb diverse approaches geprobeerd, bijv. [dit voorbeeld uit de docs](https://testing-library.com/docs/ecosystem-user-event/#input-typetime--support) en vele SO threads, maar de time input kreeg ik helaas niet aan de praat. Om die reden ben ik voor nu teruggevallen op `fireEvent`. Vervolgens heb ik vanwege consistentie ervoor gekozen om voor dit hele bestand de `fireEvent` methode aan te houden, aangezien er anders `fireEvent` en `userEvent` door elkaar heen gebruikt zouden worden.